### DISCUSSION TITLE
Rebuilding Read Models from Aggregate Snapshots

### DISCUSSION THREAD

**Discussion by fratting** [rasmus/EventFlow#908] [Q&A]
Curious to get some other thoughts on this...
We are starting to apply snapshots on our aggregates. Which is cool because we could (if desired) do some code cleanup by removing deprecated events and event upgraders and just leave the pre-snapshotted events in the event store for audit purposes without ever really needing to re-apply them. However, **that's not entirely the case when it comes to read model rebuilding** and we have to leave the deprecated events and upgraders around solely for that purpose. It would seem logical that a snapshot of an aggregate is just as much a source of truth as the events that were re-applied to produce that snapshot and with that being the case why couldn't we have a method to rebuild read models from the most recent version of a snapshot of an aggregate vs. re-application of all it's events (including deprecated events)? Is this more of a philosophical reason that events are the only source of truth and snapshots should not be treated as such? Is this just a feature not yet available with read models? Or am I just not looking in the right place in the source code to find out how it can be done?

**Comment by rasmus** [comment]
Since there aren't really a close link between read models and aggregates, there are no built-in way of rebuilding read models based on snapshots. There might be a possibility, but no something I have explored.

You could create an issue with the feature request just to make sure that the idea isn't forgotten
